source/_posts/2019-01-15-Bugsnag.md

```diff
@@ -0,0 +1,52 @@
+---
+title: Bugsnag for Vapor 3
+date: 2019-01-15 10:00:00
+tags: server,bugsnag,report,reporting,release
+authorIds: 
+- brto
+categories:
+- Vapor
+---
+
+At Nodes we have been working hard on a Vapor 3 provider for the [Bugsnag](https://www.bugsnag.com) reporting tool. Bugsnag is a reporting tool that we use extensively to help catch and fix bugs.
+
+In the new version of the library, we've been able to remove a lot of dead code, simplify the API and add support for breadcrumbs.
+
+## Migration to Vapor 3
+
+The migration from Vapor 2 to Vapor 3 was a relatively easy one for Bugsnag. The major changes were the use of `Codable` and the introduction of `Future`s.
+
+One of the double-edged-swords of Vapor 3 is the asynchronous design. The asynchronous nature of the framework makes stack traces virtually useless. Sadly, we had to remove support for CStack and stack traces.
```

Comment:
What is CStack? Or is it just me being the ignorant frontender here?

Reply:
was it the one you wrote a blog post about at some point?

Reply:
Yeah

Reply:
Maybe a link to that blog post?